src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +55,170 @@ const ActorInit Shot_Sun_InitVars = {
     (ActorFunc)ShotSun_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/ShotSun_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/ShotSun_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000020, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 30, 60, 0, { 0, 0, 0 } },
+};
+
+void ShotSun_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ShotSun* this = THIS;
+    s32 temp_v0;
+
+    // Translation: Ocarina secret occurrence
+    osSyncPrintf("%d ---- オカリナの秘密発生!!!!!!!!!!!!!\n", this->actor.params);
+    temp_v0 = this->actor.params & 0xFF;
+    if (temp_v0 == 0x40 || temp_v0 == 0x41) {
+        this->unk_1A4[0] = 0;
+        this->actor.flags |= 0x10;
+        this->actor.flags |= 0x2000000;
+        this->actionFunc = &func_80BADF0C;
```

Comment:
`&` is not required for function pointers. The same applies below.
```suggestion
        this->actionFunc = func_80BADF0C;
```